homeassistant/components/sensor/sigfox.py

```diff
@@ -0,0 +1,159 @@
+    auth = sigfox.auth
+    devices = sigfox.devices
+
+    sensors = []
+    for device in devices:
+        sensors.append(SigfoxDevice(device, auth, name))
+    add_devices(sensors, True)
+
+
+def epoch_to_datetime(epoch_time):
+    """Take an ms since epoch and return datetime string."""
+    return datetime.datetime.fromtimestamp(epoch_time).isoformat()
+
+
+class SigfoxAPI(object):
+    """Class for interacting with the SigFox API."""
+
+    def __init__(self, api_login, api_password):
+        """Initialise the API object."""
+        self._auth = requests.auth.HTTPBasicAuth(api_login, api_password)
+        if self.check_credentials():
+            device_types = self.get_device_types()
+            self._devices = self.get_devices(device_types)
+
+    def check_credentials(self):
+        """"Check API credentials are valid."""
+        url = '{}{}'.format(API_URL, 'devicetypes')
+        response = requests.get(url, auth=self._auth, timeout=10)
+        if response.status_code != 200:
+            if response.status_code == 401:
+                _LOGGER.error(
+                    "Invalid credentials for Sigfox API")
+            else:
+                _LOGGER.error(
+                    "Unable to login to Sigfox API, error code %s", str(
+                        response.status_code))
+            raise ValueError('Sigfox component not setup')
+        return True
+
+    def get_device_types(self):
+        """Get a list of device types."""
+        url = '{}{}'.format(API_URL, 'devicetypes')
+        response = requests.get(url, auth=self._auth, timeout=10)
+        device_types = []
+        for device in json.loads(response.text)['data']:
+            device_types.append(device['id'])
+        return device_types
+
+    def get_devices(self, device_types):
+        """Get the device_id of each device registered."""
+        devices = []
+        for unique_type in device_types:
+            url = '{}{}{}{}'.format(
+                API_URL, 'devicetypes/', unique_type, '/devices')
+            response = requests.get(url, auth=self._auth, timeout=10)
+            devices_data = json.loads(response.text)['data']
+            for device in devices_data:
+                devices.append(device['id'])
+        return devices
+
+    @property
+    def auth(self):
+        """Return the API authentification."""
+        return self._auth
+
+    @property
+    def devices(self):
+        """Return the list of device_id."""
+        return self._devices
+
+
+class SigfoxDevice(Entity):
+    """Class for single sigfox device."""
+
+    def __init__(self, device_id, auth, name):
+        """Initialise the device object."""
+        self._device_id = device_id
+        self._auth = auth
+        self._message_data = {}
+        self._name = '{}{}{}'.format(name, '_', device_id)
+        self._state = None
+
+    def get_last_message(self):
+        """Return the last message from a device."""
+        url = '{}{}{}{}'.format(
```

Comment:
Why you add static string to format? Please use only variables.

Reply:
OK